src/services/DiscordService.ts

```diff
@@ -0,0 +1,228 @@
+  action: string
+  color: MessageColors
+  url: string
+}
+
+function getChoices(choices: string[]): Field[] {
+  return choices.map((choice, idx) => ({
+    name: `Option #${idx + 1}`,
+    value: capitalizeFirstLetter(choice),
+  }))
+}
+
+function getPreviewText(text: string) {
+  return text.length > PREVIEW_MAX_LENGTH ? text.slice(0, PREVIEW_MAX_LENGTH) + '...' : text
+}
+
+export class DiscordService {
+  static init() {
+    if (!TOKEN) {
+      throw new Error('Discord token missing')
+    }
+
+    client.login(TOKEN)
+  }
+
+  private static get channel() {
+    if (!CHANNEL_ID) {
+      throw new Error('Discord channel ID not set')
+    }
+
+    const channel = client.channels.cache.get(CHANNEL_ID)
+    if (channel?.type !== ChannelType.GuildText) {
+      throw new Error(`Discord channel type is not GuildText ${ChannelType.GuildText}`)
+    }
+
+    return channel
+  }
+
+  private static async formatMessage({
+    title,
+    proposalType,
+    description,
+    fields: choices,
+    user,
+    action,
+    color,
+    url,
+  }: EmbedMessageProps) {
+    const fields: Field[] = []
+
+    if (!!proposalType && !!description) {
+      const embedDescription = !isGovernanceProcessProposal(proposalType)
+        ? description.split('\n')[0]
+        : getPreviewText(description)
+
+      fields.push({
+        name: proposalType.toUpperCase().replaceAll('_', ' '),
+        value: embedDescription,
+      })
+    }
+
+    if (choices.length > 0) {
+      fields.push({ name: BLANK, value: BLANK }, ...choices)
+    }
+
+    const embed = new EmbedBuilder()
+      .setColor(color)
+      .setTitle(title)
+      .setURL(url)
+      .setDescription(action)
+      .setThumbnail(DCL_LOGO)
+      .addFields(...fields)
+      .setTimestamp()
+      .setFooter({ text: 'Decentraland DAO', iconURL: DCL_LOGO })
+
+    if (user) {
+      try {
+        const profile = await profiles.load(user)
+        const hasDclProfile = !!profile && !profile.isDefaultProfile
+        const profileHasName = hasDclProfile && !!profile.name && profile.name.length > 0
+        const displayableUser = profileHasName ? profile.name! : user
+
+        const hasAvatar = !!profile && !!profile.avatar
+
+        embed.setAuthor({
+          name: displayableUser,
+          iconURL: hasAvatar ? profile.avatar?.snapshots.face256 : DEFAULT_AVATAR,
+          url: getProfileUrl(user),
+        })
+      } catch (error) {
+        console.error(`Error loading profile for user ${user}`, error)
+        embed.setAuthor({
+          name: user,
+          iconURL: DEFAULT_AVATAR,
+          url: getProfileUrl(user),
+        })
+      }
+    }
+
+    return embed
+  }
+
+  static newProposal(
+    proposalId: string,
+    title: string,
+    type: ProposalType,
+    description: string,
+    choices: string[],
+    user: string
+  ) {
+    const action = 'A new proposal has been created'
+    const embedChoices = getChoices(choices)
+    inBackground(async () => {
```

Comment:
@ncomerci we need to return a result for the inBackground successful logging.

```typescript
 .then((result) => logger.log('Completed background task', { result: JSON.stringify(result) }))
 ```

Maybe the action name, & proposal/update id